### ISSUE TITLE
Bug: 9.39.0 causes React components in conditionals to fail

### ISSUE THREAD

**Issue by larsmunkholm** [eslint/eslint#20277]
### Environment

Node version: 24.9.0
npm version: 11.6.0
Local ESLint version: 9.39.0
Global ESLint version: 9.39.0
Operating System: macOS Tahoe 26.0.1

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Eslint configuration</summary>

```js
"@typescript-eslint/strict-boolean-expressions": [
    "error",
    { allowNumber: false, allowString: false, allowNullableBoolean: true },
]
```
</details>

After the update to v9.39.0, I get 1200+ linting errors in my project, on all React components rendered with a condition shorthand like this

```js
{error != null && <Alert severity="error">{t(error)}</Alert>}
```

I know this is a plugin, so I will post this issue to them as well. ([here](https://github.com/typescript-eslint/typescript-eslint/issues/11734))
But for anyone getting the same errors, downgrading eslint to v9.38.0 helps.

### What did you expect to happen?

No errors

### What actually happened?

Lots of errors

### Link to Minimal Reproducible Example

https://bolt.new/~/sb1-fsvbwljw

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by DMartens** [comment]
Thank your for the report but the issue is already known and it is being discussed in https://github.com/eslint/eslint/issues/20272.
